viewing_party/party.py

```diff
@@ -1,23 +1,175 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if not None in (title, genre, rating):
+        movie_dict['title'] = title
+        movie_dict['genre'] = genre
+        movie_dict['rating'] = rating
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+# helper function to remove movie from watchlist
+def remove_movie_from_watchlist(user_data, movie):
+    user_data['watchlist'].remove(movie)
+
+def watch_movie(user_data, title):
+    for movie in user_data['watchlist']:
+        if title in movie['title']:
+            add_to_watched(user_data, movie)
+            remove_movie_from_watchlist(user_data, movie)
+    return user_data

```

Comment:
The helper functions are also editing the same data structure. So in this case, we can avoid the dangers* of modifying a data structure while iterating by adding a `break` once changes are made:

```suggestion
    for movie in user_data['watchlist']:
        if title in movie['title']:
            add_to_watched(user_data, movie)
            remove_movie_from_watchlist(user_data, movie)
            break
    return user_data

```

Reply:
*dangers include this scenario:

When removing an item from a list, the elements get shifted to a different index which can cause unexpected behavior (like an IndexError or the loop ending much earlier than expected)